snippets/standard/datetime/json/datetime-converter-examples/example2/Program.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Buffers;
+using System.Buffers.Text;
+using System.Diagnostics;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace DateTimeConverterExamples
+{
+    public class DateTimeConverterExample2 : JsonConverter<DateTime>
+    {
+        public override DateTime Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType != JsonTokenType.String)
+            {
+                throw new JsonException();
+            }
+
+            if (Utf8Parser.TryParse(reader.ValueSpan, out DateTime value, out _, 'R'))
+            {
+                reader.Read();
+                return value;
+            }
+
+            throw new FormatException();
+        }
+
+        public override void Write(Utf8JsonWriter writer, DateTime value, JsonSerializerOptions options)
+        {
+            Span<byte> destination = new byte[29];
+
+            bool result = Utf8Formatter.TryFormat(value, destination, out _, new StandardFormat('R'));
```

Comment:
Will this always fill the whole 29 bytes?

Maybe make it 31 and manually append the quotes at start/end (and slice on `bytesWritten + 1`)?

Then, you don't need to re-encode it as string and can write the utf-8 bytes directly:
```C#
// or something to that affect depending on how much gets written
writer.WriteStringValue(destination.Slice(0, bytesWritten + 1));
```

Reply:
> Will this always fill the whole 29 bytes?

It'll always be 29 per the standard format. I'll add a comment.

> Then, you don't need to re-encode it as string and can write the utf-8 bytes directly:
Nice, done!